datastore/google/cloud/datastore/client.py

```diff
@@ -116,28 +116,29 @@ def _extended_lookup(connection, project, key_pbs,
     while loop_num < _MAX_LOOPS:  # loop against possible deferred.
         loop_num += 1
 
-        results_found, missing_found, deferred_found = connection.lookup(
+        lookup_response = connection.lookup(
             project=project,
             key_pbs=key_pbs,
             eventual=eventual,
             transaction_id=transaction_id,
         )
 
-        results.extend(results_found)
+        # Accumulate the new results.
+        results.extend(result.entity for result in lookup_response.found)
 
         if missing is not None:
-            missing.extend(missing_found)
+            missing.extend(result.entity for result in lookup_response.missing)
```

Comment:
Yeah, source diving in the `pb2.py` generated files is seldom (never?) a good idea. The underlying proto [definition][1] would have given you the same answer. Proto definitions should always be the source of truth.

[1]: https://github.com/googleapis/googleapis/blob/1068554917fcda5facfee95c19e2aa879665316b/google/datastore/v1/datastore.proto#L93